Client/mods/deathmatch/logic/rpc/CWeaponRPCs.cpp

```diff
@@ -185,6 +186,42 @@ void CWeaponRPCs::TakeWeapon(CClientEntity* pSource, NetBitStreamInterface& bitS
     }
 }
 
+void CWeaponRPCs::TakeWeapons(CClientEntity* pSource, NetBitStreamInterface& bitStream)
+{
+    std::uint8_t count = 0;
+    if (!bitStream.Read(count))
+        return;
+
+    CClientPed* pPed = m_pPedManager->Get(pSource->GetID(), true);
+    if (!pPed)
+        return;
+
+    for (std::uint8_t i = 0; i < count; i++)
+    {
+        unsigned char weaponID, ammo, slot;
+        if (!bitStream.Read(weaponID)) continue;
```

Comment:
When bitstream returns false then we can suspect that a network packet was currupted. In this case the best we can do is to exit a function.